src/components/markdown/Document.js

```diff
@@ -2,16 +2,38 @@ import React, { PureComponent } from 'react';
 import Preview from './Preview';
 import Editor from './Editor';
 import styles from './Document.css';
+import store from '../../store';
+import { getMarkdown } from '../../selectors/selectMarkdown'; 
+import { updateMarkdown } from '../../actions/actionUpdateMarkdown';
 
 export default class Document extends PureComponent {
   state = {
-    markdown: '# Hi there'
+    markdown: '# Hi there',
+    unsubscribe: null
+  };
+
+  updateState = () => {
+    const currentMarkdown = store.getState();
```

Comment:
why call this `currentMarkdown`? Its your entire state, potentially not just markdown